### PR TITLE
Fuzzy query: Problems occurred when I used "_汉字" for fuzzy query

### PR DESCRIPTION
**Pull request by cheny-alf**
Since a Chinese character takes up three bytes, we cannot only return the index of the first byte of each Chinese character

### CODE REVIEW

File: dialect/sql/builder.go [hunk 1]
```diff
@@ -1689,11 +1689,11 @@ func escape(w string) (string, bool) {
...
 	var b strings.Builder
 	b.Grow(len(w) + n)
-	for i := range w {
+	for i, character := range w {
 		if c := w[i]; c == '%' || c == '_' || c == '\\' {
 			b.WriteByte('\\')
 		}
-		b.WriteByte(w[i])
+		b.WriteRune(character)
 	}
```
Comment (a8m) [inline dialect/sql/builder.go:1697]:
```suggestion
	for _, c := range w {
		if c == '%' || c == '_' || c == '\\' {
			b.WriteByte('\\')
		}
		b.WriteRune(c)
	}
````
Isn't this simpler?

File: dialect/sql/builder_test.go [hunk 2]
```diff
@@ -2067,6 +2067,12 @@ func TestEscapePatterns(t *testing.T) {
 		Query()
 	require.Equal(t, "UPDATE `users` SET `name` = NULL WHERE `nickname` LIKE ? ESCAPE ? OR `nickname` LIKE ? ESCAPE ? OR `nickname` LIKE ? ESCAPE ? OR LOWER(`nickname`) LIKE ? ESCAPE ?", q)
 	require.Equal(t, []any{"\\%a8m\\%%", "\\", "%\\_alexsn\\_", "\\", "%\\\\pedro\\\\%", "\\", "%\\%abcd\\%efg%", "\\"}, args)
+
+	//Chinese
```
Comment (a8m) [inline dialect/sql/builder_test.go:2071]:
```suggestion
```

Excellent test, just remove the comment. It's not needed.

**Review by a8m — APPROVED** [review]
Thanks for fixing this, @cheny-alf 🚀 

I added 2 minor style comments.

File: dialect/sql/builder_test.go [hunk 3]
```diff
@@ -2067,6 +2067,12 @@ func TestEscapePatterns(t *testing.T) {
...
+	query, args := Select("*").From(Table("dataset")).
+		Where(Contains("title", "_第一")).Query()
+	require.Equal(t, "SELECT * FROM `dataset` WHERE `title` LIKE ?", query)
+	require.Equal(t, []any{"%\\_第一%"}, args)
```
Comment (a8m) [inline dialect/sql/builder_test.go:2075]:
```suggestion
	q, args = Select("*").From(Table("dataset")).
		Where(Contains("title", "_第一")).Query()
	require.Equal(t, "SELECT * FROM `dataset` WHERE `title` LIKE ?", q)
	require.Equal(t, []any{"%\\_第一%"}, args)
```

Use the same variables from above to keep this function consistent. Thanks 🙏

**Comment by cheny-alf** [comment]
Made changes according to your suggestions 🙏 @a8m

**Pull request merged**